### PR TITLE
feature request rework user navigation to be more clearer

### PR DESCRIPTION
**Pull request by f3ltron**
**Summary**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature
- [x] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

I was thinking about re organize user navigation stuff.

Why ? 

Actually when i want to find some informations about vuepress  i feel like a bit lost.

What i propose ? 

You can clone and checkout the branch and will have example on en.

In Learn more you will now have:
- Config reference
- Api
- Create a plugin
- Create a theme
- Contribution Guide
- miscellaneous

Why ?

I wanted to regroup informations do not split them.

1) You will now have one nav for plugins and theme => Plugins / Theme  | OR | Market Place as a proposal.
 
2) As we plan to have some listing about themes and plugins community 

I thought about doing some sort listing like:
- https://events.vuejs.org/meetups/
- or https://market.nativescript.org/?tab=plugins

By doing this i splited the creation part of them into the menu before Learn more.

It will be more explicit with an img 👍 

Screen Shot 2019-09-05 at 11 26 47 PM

### CODE REVIEW

File: packages/docs/docs/.vuepress/nav/en.js [hunk 1]
```diff
@@ -4,53 +4,31 @@ module.exports = [
...
-          {
-            text: "Node",
-            link: "/api/node.html"
-          }
-        ]
+        text: "Config reference",
+        link: "/config/index.html"
+      },
+      {
+        text: "Api",
```
Comment (bencodezen) [inline packages/docs/docs/.vuepress/nav/en.js]: Should we be specific here that it is the CLI API?

Comment (f3ltron) [reply packages/docs/docs/.vuepress/nav/en.js]: not really actually i didnt do a lot of stuff i wanted first to present the idea behind

**Comment by bencodezen** [comment]
@f3ltron I think this is a great start. Thanks for doing this! I think all we need is for you to make the PR semantic and I'm happy to approve so we can merge this!

**Comment by f3ltron** [comment]
maybe i should implement the plugin and theme list before too @bencodezen ?

We just have to choose what we design we want tell me wich one your prefer:

- https://events.vuejs.org/meetups/
- or https://market.nativescript.org/?tab=plugins

**Comment by bencodezen** [comment]
@f3ltron I think the NativeScript one is a good start.

Also, if it's going to take a few days to get that done (i.e., themes and plugins list), I'm thinking we can split that out to a separate issue? The other information is good and is worth exposing to the public sooner rather than later.

**Comment by f3ltron** [comment]
I was thinking about doing a simple listing first and next reproduce nativescript design

**Comment by f3ltron** [comment]
Ok i will change that dont know why my vscode is not using the linter

**Comment by ulivz** [comment]
Cool, It would be better if we could abide by the following principles:

1. Avoid unnecessary formatted changes, so we can review the changes and track the history easily.
2. Follow the historical convention, if it needs to be changed, we can open new topics to discuss it.

**Comment by f3ltron** [comment]
@ulivz @bencodezen i have a quesiton about plugins / theme validation do we have some requirement to valid theme entering into community plugin ? 

I was thinking about :
- valid docs that explain the plugin (require)
- unit test that will validate important functions ? (optional / require)
- the plugin must come with something usefull (require)
- comes with a PR 

Have no idea about theme : @jlooper maybe you have ideas ? It could help me a lot

**Comment by f3ltron** [comment]
Ok i will need some help on default theme templates and with translations

**Comment by bencodezen** [comment]
@f3ltron I think your suggestions around official themes / plugins requiring a certain level of docs and tests is a great idea. Let's build on that in Notion so we have a template that everyone can follow when we publish it on the site.

**Comment by f3ltron** [comment]
Block for now. Waiting for on #1847

**Comment by f3ltron** [comment]
yeah i think it's relative to the last merge i will resolve them.

There is not real breaking change here just navigation rework and addition

**Comment by ulivz** [comment]
IMO at least the Config should be placed behind Guide at the same level, which is the convention of all Vue.js projects.

image

// cc @yyx990803

**Comment by f3ltron** [comment]
We wanted to introduce something more like nuxt doc. 

Like a getting started that will redirect to the config reference in some points. The idea was to point the plugins library first and the getting started.

People want to use vuepress fast and know about libs (plugin  / theme)

I am not sure if i explain it well

**Pull request closed without merging**